awx/main/tasks.py

```diff
@@ -149,6 +150,46 @@ def handle_setting_changes(self, setting_keys):
             break
 
 
+@task(bind=True, queue='tower_broadcast_all', base=LogErrorsTask)
+def handle_ha_toplogy_changes(self):
+    instance = Instance.objects.me()
+    logger.debug("Reconfigure celeryd queues task on host {}".format(self.request.hostname))
+    (instance, removed_queues, added_queues) = register_celery_worker_queues(self.app, self.request.hostname)
+    logger.info("Workers on tower node '{}' removed from queues {} and added to queues {}"
+                .format(instance.hostname, removed_queues, added_queues))
+    updated_routes = update_celery_worker_routes(instance, settings)
+    logger.info("Worker on tower node '{}' updated celery routes {} all routes are now {}"
+                .format(instance.hostname, updated_routes, self.app.conf.CELERY_ROUTES))
+
+
+@worker_ready.connect
+def handle_ha_toplogy_worker_ready(sender, **kwargs):
+    logger.debug("Configure celeryd queues task on host {}".format(sender.hostname))
+    app = Celery(broker=settings.BROKER_URL)
```

Comment:
could you get the same thing by referencing `sender.app`, as opposed to instantiating it here?

Source: stack overflow https://stackoverflow.com/a/14589445/1092940